syntax = "proto3";

package tangent.test;

enum MyEnumA {
  MyEnumA_VALUE1 = 0;
  MyEnumA_VALUE2 = 1;
  MyEnumA_VALUE3 = 2;
}

message MyMessageA {
  sint32 fieldA = 1;
  double fieldB = 2;
  uint64 fieldC = 3;
}

message MymessageB {
  reserved 1, 3, 8, 12 to 13;

  MyMessageA fieldA = 2;
  int32 fieldB = 4;
  int32 fieldF = 7;
}

message MyMessageC {
  repeated MyMessageA fieldA = 1;
  repeated int32 fieldB = 2 [packed=true];
}
